models/user.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const  { Model } = require('objection')
+const Knex = require("knex")
+
+// Initialize knex.
+const knex = Knex({
+    client: 'sqlite3',
+    useNullAsDefault: true,
+    connection: {
+        filename: 'one-sg.db3'
+    }
+});
+
+Model.knex(knex)
+
+const tableName = 'users'
+//@akshata, this is the issue. Error said:  migration failed with error: Class constructor User cannot be invoked without 'new'
+class User extends Model {
+    static get tableName () {
+        return tableName
+    }
+
+    fullName() {
+        return `${this.firstName} ${this.lastName}`
+    }
+
+    static get jsonSchema() {
+        return {
+            type: 'object',
+            required: ['email'],
+            properties: {
+                id: { type: 'integer' },
+                email: { type: 'string', minLength: 1, maxLength: 255 },
+                passwordHash: { type: 'string' },
+                firstName: { type: 'string', maxLength: 255 },
+                lastName: { type: 'string', maxLength: 255 },
+                occupation: {type: 'string', maxLength: 255}
+            }
+        }
+    }
+}
+
+module.exports = {
+    User,
+    model: User,
+    tableName
+}
```

Comment:
@herzizza88 note what you are exporting here

Reply:
the User on line 45 is the class. based on objection.js syntax , they didnt put model: User and tableName. shld i follow their convention? cause i realised that exporting user is alrdy the model itself. 

https://vincit.github.io/objection.js/guide/models.html#examples

Reply:
Makes sense - but what you export from there, is upto the requirements of what you particular conventions :)